src/backend/opencl/magma/laset.cpp

```diff
@@ -7,6 +7,50 @@
  * http://arrayfire.com/licenses/BSD-3-Clause
  ********************************************************/
 
+/***********************************************************************
+ * Based on MAGMA library http://icl.cs.utk.edu/magma/
+ * Below is the original copyright.
```

Comment:
Why don't we add the Magma license to the LICENSES directory instead of pasting the full licenses in each of the files? For the example/random123, its only a single file, so it doesn't make that much of a difference.